hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java

```diff
@@ -179,10 +181,10 @@ public PreHandleContext requireKey(@NonNull final HederaKey key) {
      * @throws PreCheckException if the key is null or empty
      */
     @NonNull
-    public PreHandleContext requireKeyOrThrow(
-            @Nullable final HederaKey key, @NonNull final ResponseCodeEnum responseCode) throws PreCheckException {
+    public PreHandleContext requireKeyOrThrow(@Nullable final Key key, @NonNull final ResponseCodeEnum responseCode)
+            throws PreCheckException {
         requireNonNull(responseCode);
-        if (key == null || key.isEmpty()) {
+        if (key == null || !isValid(key)) {
```

Comment:
As noted above, I'd suggest make `isValid(null)` just return `false` and simplify this condition.